helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/SoftConstraint.java

```diff
@@ -24,8 +24,9 @@
 import org.apache.helix.controller.rebalancer.waged.model.ClusterContext;
 
 /**
- * Evaluate a partition allocation proposal and return a score within the normalized range.
- * A higher score means the proposal is more preferred.
+ * The "soft" constraint evaluates the potential assignment by giving it score
```

Comment:
the potential assignment -> the optimality of an assignment on a scale of 0 to maxScore.
giving it "a" score